### PR TITLE
RPP Magnitude on HOST and HIP

### PR DESCRIPTION
**Pull request by r-abishek**
- Adds magnitude functionality to HOST and HIP with support for U8/F16/F32/I8 depths and NCHW/NHWC layouts.
- Adds relevant QA tests, unit tests, performance tests

### CODE REVIEW

**Comment by kiritigowda** [comment]
@r-abishek  - need to resolve conflicts to run CI

File: src/modules/hip/kernel/magnitude.hpp [hunk 1]
```diff
@@ -0,0 +1,262 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__device__ void magnitude_hip_compute(uchar *srcPtr, d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8)
```
Comment (rrawther) [inline src/modules/hip/kernel/magnitude.hpp:4]: please make them templated functions as discussed

File: utilities/test_suite/REFERENCE_OUTPUT/magnitude/magnitude_u8_Tensor_PKD3.csv
Comment (rrawther) [inline utilities/test_suite/REFERENCE_OUTPUT/magnitude/magnitude_u8_Tensor_PKD3.csv:1]: please combine all csvs into single one

**Review by rrawther — CHANGES_REQUESTED** [review]
please address review comments

**Comment by rrawther** [comment]
@AryanSalmanpour : Can you please review the HIP kernels

**Comment by r-abishek** [comment]
@kiritigowda Ubuntu 20.04 machine has no space on device left, so CI fails with that error for this PR.

**Comment by kiritigowda** [comment]
@r-abishek - conflicts need to be resolved.

**Comment by r-abishek** [comment]
@kiritigowda Conflicts resolved

**Pull request merged**